src/components/Profile/addresses/AddressCard.tsx

```diff
@@ -0,0 +1,174 @@
+import { Card, CardContent, CardFooter } from '@/components/ui/card';
+import { Pencil, Trash, Home } from 'lucide-react';
+import { Switch } from '@/components/ui/switch';
+import { Address } from '@commercetools/platform-sdk';
+import { Button } from '@/components/ui/button';
+
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger
+} from '@/components/ui/dialog';
+import { Formik } from 'formik';
+import { AddressFieldsValues } from './AddressSection';
+import AddressFields from './AddressFields';
+import { useAuth } from '@/context/AuthContext';
+import { useState } from 'react';
+import { toast } from 'sonner';
+import { updateUserAddresses } from './updateAddress';
+import { handleErrors } from '@/components/Register/registerUtils';
+import * as Yup from 'yup';
+
+type Props = {
+  address: Address;
+  isDefault: boolean;
+  onDelete: () => void;
+  onSetDefault: () => void;
+};
+
+type AddresFields = Omit<AddressFieldsValues, 'type'>;
+
+const AddressCard = ({ address, isDefault, onDelete, onSetDefault }: Props) => {
+  const { user, refreshUser } = useAuth();
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  if (!address.id) {
+    toast.error('No address to change.');
+
+    return;
+  }
+
+  const addressId = address.id;
+  const initialValues: AddresFields = {
+    country: address?.country ?? '',
+    city: address?.city ?? '',
+    streetName: address?.streetName ?? '',
+    postalCode: address?.postalCode ?? '',
+    isDefault: false
+  };
+
+  const adressValidationSchema = Yup.object({
+    country: Yup.string()
+      .matches(/^[A-Za-zÀ-ÿ' -]+$/, 'Field must only contain letters')
+      .required('Country is required'),
+    city: Yup.string()
+      .matches(/^[A-Za-zÀ-ÿ' -]+$/, 'Field must only contain letters')
+      .required('City is required'),
+    streetName: Yup.string()
+      .matches(/^[A-Za-zÀ-ÿ0-9\s,'/.-]{2,}$/, 'Field must contain letters & numbers')
+      .required('Street is required'),
+    postalCode: Yup.string()
+      .matches(/^[A-Za-z0-9\s-]{3,10}$/, 'Invalid postal code format')
+      .required('Postal code is required')
+  });
+
+  const handleEdit = async (values: Partial<AddressFieldsValues>) => {
+    if (!user) {
+      toast.error('User info is missing.');
+
+      return;
+    }
+
+    const changes: Partial<AddressFieldsValues> = { ...values };
```

Comment:
`changes` is already created from `values`. Making smth like this `changes.country = values.country` seems useless.